instagramapp/src/components/Login/Login.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { Form } from 'reactstrap';
+import { Input } from 'reactstrap';
+import { Button } from 'reactstrap';
+
+class Login extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            username: '',
+            password: ''
+        }
+    }
+
+    toggleComponents = () => {
+        if (!this.state.showFirst) {
+          localStorage.setItem('showFirst', true);
+        } else {
+          localStorage.removeItem('showFirst');
+        }
+        this.setState({ showFirst: !this.state.showFirst });
```

Comment:
im not really sure what were doing with this 'showfirst' variable, what we can do to get the HOC to work is have a if statement in the withAuthenticate component and just check if there is a username in our local storage and render login or postPage based on that.